prometheus-openstack-exporter

```diff
@@ -616,29 +617,34 @@ class Nova:
             elif not isinstance(cpu_info, dict):
                 cpu_info = json.loads(cpu_info)
                 arch = cpu_info["arch"]
-            label_values = [
-                config["cloud"],
-                host,
-                self.aggregate_map.get(host, "unknown"),
-                self.services_map[host],
-                arch,
-            ]
-            # Disabled hypervisors return None below, convert to 0
-            vms.labels(*label_values).set(squashnone(h["running_vms"]))
-            vcpus_total.labels(*label_values).set(squashnone(h["vcpus"]))
-            vcpus_used.labels(*label_values).set(squashnone(h["vcpus_used"]))
-            mem_total.labels(*label_values).set(squashnone(h["memory_mb"]))
-            mem_used.labels(*label_values).set(squashnone(h["memory_mb_used"]))
-            disk_total.labels(*label_values).set(squashnone(h["local_gb"]))
-            disk_used.labels(*label_values).set(squashnone(h["local_gb_used"]))
-
-            if config.get("schedulable_instance_size", False):
-                schedulable_instances.labels(*label_values).set(
-                    self._get_schedulable_instances(h)
-                )
-                schedulable_instances_capacity.labels(*label_values).set(
-                    self._get_schedulable_instances_capacity(h)
-                )
+
+            for agg in self.prodstack["aggregates"]:
+                agg_key = host + "_" + agg["name"]
+                if self.aggregate_map.get(agg_key) is None:
+                    continue
```

Comment:
What if there are no aggregates?  Wouldn't the metric be skipped in this case?  The old behaviour would have an `unknown` aggregate.

Reply:
Please check the update - this should preserve the old behaviour.

Reply:
Thanks :)